### ISSUE TITLE
Problème sur le décompte des heures facturées

### ISSUE THREAD

**Issue by sdutertre** [bgo-bioimagerie/platformmanager#140]
Bonjour,
je fais les facturations pour la période avril-aout 2020 et le nombre d’heures facturées ne correspond pas à aux nombres d'heures réalisées (sur le calendrier de réservation). Cela concerne l'utilisateur #809 et la réservation du Zeiss inverse IMPACCell sur la période du 01/04/2020 au 31/08/2020 (143 h facturées et je ne vois que 70h de réservées sur cette période).  

Merci 
Stéphanie

**Comment by osallou** [comment]
sur quelle plateforme?

**Comment by osallou** [comment]
ok j'ai trouvé, je vais investiguer

**Comment by osallou** [comment]
par contre je ne vois pas de réservation sur la période pour l'utilisateur 809 sur cet équipement

**Comment by sdutertre** [comment]
bizarre. L'identifiant est mcuvellier

**Comment by sdutertre** [comment]
ah, je sais. J'ai fait la facturation. Est ce que cela efface les réservations à facturer? Je peux annuler la facturation pour cette équipe.

**Comment by osallou** [comment]
cela n'efface rien, je regarde les réservations ds la base de donnée, et je ne trouve pas cet utilisateur du tout sur les resources IMPACCell, et qd je regarde le calendrier sur l'interface je ne le vois pas non plus (zeis inverse, resource id 66)

**Comment by osallou** [comment]
ou si il les efface, il faut que je vois où il les mets....

**Comment by sdutertre** [comment]
Désolé Olivier, c'est pas la bonne plateforme. Tu dois regarder sur MRIc-photonics. On gère les réservations du microscope d'Impacell pour les utilisateurs autonomes.

**Comment by osallou** [comment]
ok, mais pour cet utilisateur je ne vois que des résa pour Confocal SP5 et Multiphoton après le premier avril

**Comment by osallou** [comment]
je comprends un peu mieux, la réservation a été effectué par qq d'autre (#158), de chez vous je pense, l'info du "client" est ds un autre champ

**Comment by osallou** [comment]
je trouve donc 70h sur le reverse, et environ 6h sur d'autres équipements

**Comment by sdutertre** [comment]
oui mais c'est 173h de facturées

**Comment by osallou** [comment]
faut que je trouve où, ce n'est pas parce qu'il y a des heures de nuit?

**Comment by sdutertre** [comment]
en extrayant les réservations, pour cette personne j'ai bien deux réservations dont une qui n'est plus sur l'agenda. 

Pleins Champs | Zeiss inverse ImPACcell |   | debut : 17/07/2020 à 11:00fin : 20/07/2020 à 09:00durée : 4200 minutes |   | Défaut | mcuvellier

Pleins Champs | Zeiss inverse ImPACcell |   | debut : 31/07/2020 à 10:00fin : 03/08/2020 à 11:00durée : 4380 minutes |   | Défaut | mcuvellier

Est ce que tu penses qu'il facture les réservations qui ont été annulées?

**Comment by osallou** [comment]
je n'ai pas le détail de ce qu'il calcule mais il semble compter 2 résa (facture 1878 si je ne me trompe)

28h de jour puis 96h de jour pour la resource Zeiss inverse

mais je n'ai pas de détail de comment il est arriver là, il faudrait que je regénère la facture avec des logs pour savoir

**Comment by osallou** [comment]
autant pour moi, 3 fois:

21h au prix unitaire de 4.9 , 28h et 96h au prix de 2.45

**Comment by sdutertre** [comment]
je vais essayer de générer une facture sur une semaine avec un utilisateur réservant et déréservant.

**Comment by osallou** [comment]
pour l'instant il n'y a pas de logs (il faut modifier le code) donc pas la peine

**Comment by osallou** [comment]
la resource dans la facture ne devrait apparaitre qu'une fois (peu importe le nb de résa), et là elle apparait 3 fois, qq chose n'est pas normal

**Comment by sdutertre** [comment]
J'ai juste fait un test avec l'utilisateur 897. C'est mon utilisateur test. J'ai surtout peur qu'il me facture des heures déréservées. J'ai donc réservé 2 créneaux sur la semaine du 1er septembre puis déreservé un créneau. Bonne nouvelle, il ne me facture que le créneau réservé.

**Comment by sdutertre** [comment]
pour la facture, je pense que c'est normal parce qu'on a des heures jours, nuit et week-end

**Comment by osallou** [comment]
pas d'après le code... :-(  c'est bien la facture 1878?
il ne montre que des heures de jours

**Comment by sdutertre** [comment]
J'ai pas le même numéro de facture mais je peux te l'envoyer par mail, si tu veux. 

2 points: on a un bug mais je pensais que c'était plutôt un problème d'affichage où il me met toutes les réservations nuit et week-end sous le nom générique DMRXA (cela va être compliqué à expliquer par écrit). je pourrait refaire un autre ticket. 

deuxième point important; je viens de me rendre compte que toutes  les heures apparaissent dans le calendrier mais uniquement en affichage semaine et pas mois. Ce matin, j'ai vérifié les réservations sous le mode d'affichage mois pour cette ressource. Je n'ai donc pas de problèmes pour cette utilisatrice. Pour moi, les heures sont raccord. On a donc juste un problème d'affichage en mode mois.

**Comment by osallou** [comment]
ahh... je trouve 143h sur la resource inverse... avec les dates auxquelles tu faisais référence (17 et 31 juillet)
toutes 2 ont été "modifiées" le 01/09, mais aucune n'est annulée

**Comment by osallou** [comment]
poru 'laffichage mois, ce n'est pas parce qu'il y a "trop" de résa pour le mm jour?

**Comment by sdutertre** [comment]
non, je ne crois pas. C'est peut-être parce qu'on est à cheval sur juillet et aout.
